add initial text for sections Operations, Logical Resource Organization, and REST Binding (#33)
w3c/lws-protocol: spec/create-resource.md

The *create resource* operation requests the creation of a new resource on the server.   The resource can be a non-container (data) resource or a container (collection) resource.  The operation can either let the server assign a new identifier within a target container (like creating a new file in a folder), or use a client-specified identifier (creating or overwrite at a specific path).

**Inputs:**

* **Target location:** Either a *target container identifier* (when the client wants the server to assign a name within that container) **or** a *full target resource identifier* (when the client specifies the exact desired identifier for the new resource).

> * **Resource content:** The content to store in the resource (binary or text). This is optional if creating an empty container or an empty resource.

Reviewer: ```suggestion
* **Resource content:** The content to store in the resource (binary or text). This is optional to allow creation of an empty container or an empty resource.
```

> * **Suggested name:** An optional hint for naming (if the server will assign the identifier). For example, the client may propose a filename or resource name, which the server may use or adjust.

Reviewer: ```suggestion
* **Suggested name:** An optional hint for naming (when the server will assign the identifier). For example, the client may propose a filename or resource name, which the server may use or adjust.
```

> * **Media type:** The MIME type or format of the content (e.g., `text/plain, application/json`). This helps the server understand how to store and serve the content. It may be mandatory for some protocols when content is provided (and may be omitted if creating an empty container with no content).

Reviewer: ```suggestion
* **Media type:** The MIME media type or format of the content (e.g., `text/plain`, `application/json`). This helps the server understand how to store and serve the content. It may be mandatory for some protocols when content is provided (and may be omitted if creating an empty container with no content).
```

**Behavior:**

> * If a *target container* is provided, the server will create a new resource as a member of that container. The server determines the new resource’s identifier (often incorporating the suggested name if provided and allowed). The new resource is then added to the specified container’s membership.

Reviewer: ```suggestion
* If a *target container* is provided, the server will create a new resource as a member of that container. The server determines the new resource’s identifier (often incorporating the suggested name, if provided and allowed). The new resource is then added to the specified container’s member list.
```

* If a *full target identifier* (including the desired name/path) is provided, the server will attempt to create a resource at that exact identifier. If a full target identifier is provided and a resource already exists at that identifier, the server **MUST** fail with a Conflict response, unless the client explicitly indicates that overwriting is acceptable.

* The server **MAY** enforce additional constraints on creation. For example, it may have size limits, quota checks, or restrictions on allowed media types or file extensions. If any such constraint is violated (e.g., the content is too large or an unsupported type), the server will reject the operation with an appropriate error.

> * On successful creation, the new resource is persistently stored and linked into the logical container hierarchy. The server will assign an identifier if needed and make the resource available for future operations. The identifier (e.g., a URI or path) of the newly created resource **MUST** be returned to the client (so the client knows how to refer to it). The creation is atomic – if any step fails, no new resource is created (or any partial resource is cleaned up).

Reviewer: ```suggestion
* Upon successful creation, the new resource is persistently stored and linked into the logical container hierarchy. The server will assign an identifier, if needed, and make the resource available for future operations. The identifier (e.g., a URI or path) of the newly created resource **MUST** be returned to the client (so the client knows how to refer to it). The creation is atomic — if any step fails, no new resource is created (or any partial resource is cleaned up).
```

**Possible Responses:** *(on the abstract operation level, not tied to a specific protocol)*

* **Created:** The operation succeeded in creating a new resource. The response includes the new resource’s identifier and possibly a minimal representation or metadata (e.g., an ETag or version number).

* **Bad Request:** The request was malformed or violated constraints (e.g., missing required inputs, invalid identifier, disallowed media type, or size limits exceeded). The resource was not created.

* **Not Permitted:** The client is not authorized to create a resource in the target location.

> * **Conflict:** A resource at the target identifier already exists and the server does not allow overwrite (or a naming conflict occurred). No new resource was created.

Reviewer: ```suggestion
* **Conflict:** A resource at the target identifier already exists and the server does not allow overwrite (or a naming conflict occurred). No new resource was created, and any existing resource remains unchanged.
```

> * **Not Found:** (If applicable) The specified target container does not exist or cannot be found. This could also be treated as a type of Bad Request or Not Permitted depending on the cause (e.g., if the container is missing vs. the client lacking access).

Reviewer: ```suggestion
* **Not Found:** (If applicable) The specified target container does not exist or cannot be found. This could also be treated as a type of Bad Request or Not Permitted, depending on the cause (e.g., if the container is missing or the client lacks access permission).
```
